projects/js-toolkit/docs/manuals/liferay-cli.md

```diff
@@ -53,6 +57,52 @@ Once the project is created, you may change to its folder and run
 `npm|yarn install` to install the dependencies, then invoke the supported
 actions (like `build`, `deploy`, etc.) with `npm|yarn run <command>`.
 
+#### Project types
+
+Currently the following project types are supported.
+
+##### Liferay Platform Project
+
+The typical project from JS Toolkit that creates a deployable JAR file
+implementing a Widget (portlet) that can be added to any page.
+
+The project may support different features like
+[localization](./features/localization.md) and
+[configuration](./features/configuration.md), and it follows the
+[JavaScript portlet entry point](./reference/js-portlet-entry-point.md)
+contract.
+
+##### Liferay Remote App
+
+Remote Apps implement a
+[Custom Element](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements)
+that can be registered with Liferay DXP in the Remote Apps configuration. This
+generates a new Widget that is rendered using the custom element, instead of
+the old [JavaScript portlet entry point](./reference/js-portlet-entry-point.md)
+method.
+
+Currently only [React](https://reactjs.org/) framework is supported[^1], and you
+can decide whether to use the React copy Liferay DXP uses at runtime, or bring
+your own copy inside the generated JavaScript file. You can do so when selecting
+the target platform when asked:
+
+```sh
+? Which will be your target platform? (Use arrow keys)
+❯ Liferay Portal CE (not sharing platform's packages)
+  Liferay Portal CE 7.4
+  Liferay DXP 7.4
+```
+
+For example, if you select `Liferay Portal CE (not sharing platform's packages)`
+your own copy of React will be bundled inside the generated JavaScript file
+which will be completely isolated from the one Liferay DXP uses.
```

Comment:
How exactly is it isolated? I'm sure I should have known this already, but I assumed multiple versions of react would break.

Unless this is a custom elements thing that isolates the js

Reply:
It is isolated by webpack. I mean, a copy of React ends up inside the index.js file generated when building the project.